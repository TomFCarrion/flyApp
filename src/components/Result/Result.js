import React from 'react';


const Result= () => {
    
    return (
        <h1>Search Your Flight </h1>
    );
};
export default Result;

