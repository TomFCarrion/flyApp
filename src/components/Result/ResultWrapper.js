import React from 'react';


const ResultWrapper = () => {
    
    return (
        <h1>Search Your Flight </h1>
    );
};
export default ResultWrapper;

