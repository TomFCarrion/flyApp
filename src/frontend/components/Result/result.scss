.card {
  box-sizing: border-box;
  background-color: white;
  padding: 1rem;
  margin: 1rem;
  border-radius: 10px;
  height: 300px;
  font-family: 'Open Sans', sans-serif;
}

.error {
  text-align: center;
}

.error-message {
  padding: 1rem;
  color: red;
}

.upper-item {
  text-align: center;
  &.airline {
    font-family: 'Domine', serif;
    color: #f3b305;
  }
}
.upper,
.middle,
.bottom {
  display: flex;
  justify-content: space-between;
  height: 33%;
}

.middle {
  border-bottom: 2px dashed gray;
  text-align: center;
}

.bottom {
  padding-top: 1rem;
}

.price {
  font-weight: bolder;
  font-size: 2rem;
  color: #ff715d;
}

.price-text {
  color: gray;
  float: left;
}

.dot {
  transform: translateX(-30px) translateY(-15px);
  height: 25px;
  width: 25px;
  background-color: #98cadb;
  border-radius: 50%;
  display: inline-block;
  position: relative;
  &.right {
    right: -100.5%;
  }
}

.navBack {
  border: solid white;
  border-width: 0 3px 3px 0;
  display: inline-block;
  padding: 5px;
  transform: rotate(135deg);
  -webkit-transform: rotate(135deg);
  margin: 1rem;
}
