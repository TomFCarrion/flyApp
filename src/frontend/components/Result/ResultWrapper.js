import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { FETCH_FLIGHTS } from '../../actions/actionTypes';
import Result from './Result';
import axios from 'axios';

const ResultWrapper = () => {
  const data = useSelector((state) => state.flightsReducer);
  const from = useSelector((state) => state.infoReducer.from);
  const fromId = useSelector((state) => state.infoReducer.fromId);
  const to = useSelector((state) => state.infoReducer.to);
  const toId = useSelector((state) => state.infoReducer.toId);
  const startDate = useSelector((state) => state.infoReducer.startDate);
  const endDate = useSelector((state) => state.infoReducer.endDate);
  const dispatch = useDispatch();

  const options = {
    method: 'GET',
    url: `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/browseroutes/v1.0/US/USD/en-US/${fromId}/${toId}/${startDate}`,
    params: { inboundpartialdate: endDate },
    headers: {
      'x-rapidapi-key': '08eb576b39mshaa3eb3f0249657ep1b15f7jsn598ba72db2e4',
      'x-rapidapi-host':
        'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
    },
  };

  function fetchData() {
    return (dispatch) => {
      axios
        .request(options)
        .then((res) =>
          dispatch({
            type: FETCH_FLIGHTS,
            data: res.data,
          }),
        )
        .then((res) => console.error(res))
        .catch((error) => {
          console.error(error);
        });
    };
  }

  const parseData = () => {
    let carriers = [];
    let quotes = [];
    let parsedCarriers = [];

    if (data?.data?.Carriers?.length) {
      carriers = data.data.Carriers;
      quotes = data.data.Quotes;

      for (let i = 0; i < carriers.length; i++) {
        for (let j = 0; j < quotes.length; j++) {
          if (carriers[i].CarrierId === quotes[j].OutboundLeg.CarrierIds[0]) {
            parsedCarriers.push({
              id: carriers[i].CarrierId,
              name: carriers[i].Name,
              price: quotes[j].MinPrice,
            });
          }
        }
      }
    }

    console.log(parsedCarriers);
    return parsedCarriers;
  };

  useEffect(() => {
    dispatch(fetchData());
  }, []);

  return (
    <div>
      <Link to="/">
        <div className="navBack"> </div>
      </Link>

      <ul>
        { parseData().map( ticket => 
          <Result
            key ={ ticket.id }
            from ={ from }
            to = { to }
            airlineName = { ticket.name }
            flightNumber = { ticket.id }
            ticketPrice = { ticket.price }    
            date = { startDate }
          />)
        }
      </ul>
    </div>
  );
};

export default ResultWrapper;
