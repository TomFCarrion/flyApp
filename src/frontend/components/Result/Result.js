import React from 'react';
import './result.scss';
import moment from 'moment';

const Result = ({
  from = '',
  to = '',
  airlineName = '-',
  flightNumber = '0000',
  ticketPrice = '-',
  date = '-',
}) => {
  const momentDate = moment(date).format('MMM-DD');
  return (
    <div className='Card'>
      <div className='upper'>
        <span className='upper-item'>{from}</span>
        <span className='upper-item airline'>{airlineName}</span>
        <span className='upper-item'>{to}</span>
      </div>
      <div className='middle'>
        <div>
          <div>Depart:</div>
          <span />
          <div>{momentDate}</div>
        </div>
        <div>
          <div>Flight number:</div>
          <div>{flightNumber}</div>
        </div>

      </div>
      <div className='dot' />
      <div className='dot right' />
      <div className='bottom'>
        <span className='price-text'> Ticket Price</span>
        <span />
        <span className='price'>
          {' '}
          {`$ ${ticketPrice}`}
        </span>
      </div>
    </div>
  );
};
export default Result;
