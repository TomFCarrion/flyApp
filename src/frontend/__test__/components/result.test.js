import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Result from '../../components/Result/Result';
import ProviderMock from '../../__mocks__/providerMock';

describe(' <Result/>', () => {
  test('Result component render', () => {
    const result = shallow(
      <ProviderMock>
        <Result />
      </ProviderMock>,
    );
    expect(result.length).toEqual(1);
  });
});

describe('Result snapshot', () => {
  test('Check Result snapshot', () => {
    const searchWrapper = create(
      <ProviderMock>
        <Result />
      </ProviderMock>,
    );
    expect(searchWrapper.toJSON()).toMatchSnapshot();
  });
});
