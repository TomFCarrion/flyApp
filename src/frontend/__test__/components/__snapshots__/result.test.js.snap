// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result snapshot Check Result snapshot 1`] = `
<div
  className="Card"
>
  <div
    className="upper"
  >
    <span
      className="upper-item"
    >
      
    </span>
    <span
      className="upper-item airline"
    >
      -
    </span>
    <span
      className="upper-item"
    >
      
    </span>
  </div>
  <div
    className="middle"
  >
    <div>
      <div>
        Depart:
      </div>
      <span />
      <div>
        Invalid date
      </div>
    </div>
    <div>
      <div>
        Flight number:
      </div>
      <div>
        0000
      </div>
    </div>
  </div>
  <div
    className="dot"
  />
  <div
    className="dot right"
  />
  <div
    className="bottom"
  >
    <span
      className="price-text"
    >
       Ticket Price
    </span>
    <span />
    <span
      className="price"
    >
       
      $ -
    </span>
  </div>
</div>
`;
